astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py

```diff
@@ -163,3 +165,15 @@ def test_get_position():
     assert ax.coords[1].get_ticklabel_position() == ["r", "l"]
     assert ax.coords[0].get_axislabel_position() == ["t"]
     assert ax.coords[1].get_axislabel_position() == ["r"]
+
+
+def test_deprecated_getters():
+    fig = plt.figure()
+    ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8], aspect="equal")
+    helper = CoordinateHelper(parent_axes=ax)
+
+    with warnings.catch_warnings():
```

Comment:
are all 3 `assert` statement expected to emit a warning ? in general, we shouldn't be filtering these in test, but instead use `pytest.warns` to mark these warnings as *expected*. It's also best to keep the code under such contexts a minimal as possible, so only the lines that do emit the warning exist in that block.

Reply:
Yes I was going to make the same comment - we should have a separate pytest.warns context for each of the three lines to make sure we are emitting the deprecation warning.

Reply:
Yes, each of the three `assert` statements should emit. I've put each one into a separate `pytest.warns` context as suggested.